rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -89,25 +113,28 @@ void SequentialWriter::open(
   }
 
   const auto storage_uri = format_storage_uri(base_folder_, 0);
-
   storage_ = storage_factory_->open_read_write(storage_uri, storage_options.storage_id);
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
 
-  if (max_bagfile_size_ != 0 &&
-    max_bagfile_size_ < storage_->get_minimum_split_file_size())
+  if (storage_options.max_bagfile_size != 0 &&
+      storage_options.max_bagfile_size < storage_->get_minimum_split_file_size())
   {
-    throw std::runtime_error(
-            "Invalid bag splitting size given. Please provide a different value.");
+    throw std::invalid_argument{
+      "Invalid bag splitting size given. Please provide a different value."};
   }
 
-  init_metadata();
+  init_compression(compression_options);
+  init_metadata(compression_options);
 }
 
 void SequentialWriter::reset()
 {
   if (!base_folder_.empty()) {
+    if (compressor_ && compression_mode_ == CompressionMode::FILE) {
```

Comment:
Nitpick: You didn't check for `compressor_ != nullptr` on [line 202](https://github.com/ros2/rosbag2/pull/250/files#diff-07603c1ee210fdbba6c3090284fc556dR202), why be inconsistent here? I think just don't have that check. If `compressor_ == nullptr` when the `compression_mode_ == FILE`, just let the assertion in `compress_last_file()` fire.

Reply:
If the sequential writer fails to open because of bad options (ex. unable to open storage or bad splitting options) then `reset()` will be called without `compressor_` being initialized and the program will fail.